Distribution Specs (v3) - Runtime Base Image Distribution Spec (#270)
buildpacks/spec: distribution.md

  - [Distribution API Version](#distribution-api-version)
  - [Artifact Format](#artifact-format)
    - [Buildpackage](#buildpackage)

> <<<<<<< HEAD
>     - [Lifecycle](#lifecycle)
> =======
>     - [Run Image](#run-image)
> >>>>>>> runtime base image distribution spec

Reviewer: ```suggestion
    - [Lifecycle](#lifecycle)
    - [Run Image](#run-image)
```

Author: Ahh my bad, forgot to remove those.

The buildpack ID and version MUST match a buildpack provided by a layer blob.

For a buildpackage to be valid, each `buildpack.toml` describing a buildpack implementation MUST have all listed targets.

### Run Image

The following defines how a run image SHOULD be packaged for distribution as an OCI Image. The run image is the component that provides the base from which app images are built.

The image configuration refers to the OCI Image configuration as mentioned [here](https://github.com/opencontainers/image-spec/blob/main/config.md#properties).

#### Image Configuration

The Run Image MUST contain the following configurations:

> * Image Config's `config.User` field MUST be set to a non-root user with a writable home directory.

Reviewer 2: Do we want to say that the user SHOULD be different from the build image's user?

> * Image Config's `config.User` field MUST be set to a non-root user with a writable home directory and a **DIFFERENT** user [†](README.md#operating-system-conventions)UID/[‡](README.md#operating-system-conventions)SID as the build image.

Reviewer 2: ```suggestion
* Image Config's `config.User` field MUST be set to a non-root user with a writable home directory. The user SHOULD be a **DIFFERENT** user [†](README.md#operating-system-conventions)UID/[‡](README.md#operating-system-conventions)SID from that specified in the build image.
```

* Image Config's `os` field MUST be set to the underlying operating system used by the run image.
* Image Config's `architecture` field MUST be set to the underlying operating system used by the run image.

The Run Image SHOULD contain the following configurations:

* Image Config's `variant` field SHOULD be set to the underlying architecture variant.

#### Environment Variables

The Run Image MUST contain the following Environment Variables:

* Image Config's `config.Env` field MUST have the environment variable `CNB_USER_ID` set to the user UID/SID of the user specified in the `User` field.

> * Image Config's `config.Env` field MUST have the environment variable `CNB_GROUP_ID` set to the primary group GID/SID of the user specified in the `User` field.

Author: Yep, I have added the footnotes at the required places!

* Image Config's `config.Env` field MUST have the environment variable `PATH` set to a valid set of paths or explicitly set to empty (`PATH=`).

#### Labels

The Run Image SHOULD contain the following Labels on the image configuration:

| Label             | Description
| --------          | --------
| `io.buildpacks.distribution.name` |  A string denoting the Operating System distribution
| `io.buildpacks.distribution.version` | A string denoting the Operating System distribution version

> | `io.buildpacks.id` | <Target ID>

Reviewer: ```suggestion
| `io.buildpacks.id` | `<Target ID>`
```

Where,

> `<Target ID>` is an identifier specified on the runtime base image that must be provided to buildpacks as `CNB_TARGET_ID` during the build process.

Reviewer: The env var `CNB_TARGET_ID` is part of the buildpack API. I would not mention it here.

What if we change this sentence to explain what it's intended to be used for instead of how it's provided to the buildpack.

Here's my swing at it:

```suggestion
`<Target ID>` is an identifier specified on the runtime image that MAY be used to apply target-specific logic.
```

I explicitly used **MAY** here because it's irrelevant to **this spec** whether the ID is used at all. I envision that the _**MUST**_ requirement would be part of the buildpack API.

@natalieparellano / @samj1912 any contradictory opinions or alternative suggestions?
